models/reviews.model.js

```diff
@@ -76,13 +76,34 @@ exports.selectReviewsById = (review_id) => {
     });
 };
 
-exports.selectCommentsByReviewId = (review_id) => {
-  const query = `SELECT * FROM comments WHERE review_id = $1;`;
-  const columns = [review_id];
+exports.selectCommentsByReviewId = (id) => {
+  const query = `SELECT * FROM comments WHERE review_id = $1 ORDER BY created_at;`;
 
-  return Promise.all([db.query(query, columns), checkIdExists(review_id)]).then(
-    (result) => {
-      return result[0].rows;
+  return db.query(query, [id]).then((result) => {
+    if (result.rows.length === 0) {
+      return checkExists("reviews", "review_id", id);
+    } else {
+      return result.rows;
     }
-  );
+  });
+};
+
+exports.insertComment = async (id, comment) => {
+  const { review_id } = id;
+  const { username, body } = comment;
+
+  const queryStr = `INSERT INTO comments(author, body, review_id)
+  SELECT $1, $2, (SELECT review_id FROM reviews WHERE review_id = $3::INT)
+  FROM comments
+  RETURNING*;`;
```

Comment:
This select is giving you a false positive here. Its not actually inserting the values as far as i know. You'll want this to look a bit more like
INSERT INTO comments(x,y,z) VALUES($1, $2, $3) RETURNING*;